tetris.py

```diff
@@ -132,6 +144,10 @@ def new_block(self):
             self.current_block = self.upcoming_block
         self.upcoming_block = random.choice(list(blocks.values()))
```

Comment:
In `new_block()`, blocks can't be just lists of coordinates anymore. The shape letter, such as `"L"`, has to be stored into a block too. I'm imagining something like this:

```suggestion
        shape_letter = random.choice(list(blocks.keys()))
        self.upcoming_block = {coord: shape_letter for coord in blocks[shape_letter]}
```

For example, if `shape_letter` happens to be `"L"`, this code would set `self.upcoming_blocks` to

```
{(x - 1, y): "L", (x, y): "L", (x + 1, y): "L", (x + 1, y + 1): "L"}
```
instead of just
```
[(x - 1, y), (x, y), (x + 1, y), (x + 1, y + 1)]
```